test/integration/http_integration.cc

```diff
@@ -357,16 +357,24 @@ HttpIntegrationTest::waitForNextUpstreamRequest(const std::vector<uint64_t>& ups
   absl::optional<uint64_t> upstream_with_request;
   // If there is no upstream connection, wait for it to be established.
   if (!fake_upstream_connection_) {
-
     AssertionResult result = AssertionFailure();
-    for (auto upstream_index : upstream_indices) {
+    int upstream_index = 0;
```

Comment:
I wonder if the test failures are related to the fact that the original only tests upstreams with the given indices, while your version always checks all upstreams? Also, the original is much more likely to return the first member of upstream_indices with a request than the optimized version.

Reply:
Thanks! I think you are right. I should iterate over the indexes which is passed. An embarrassing mistake.